src/RestApi.ts

```diff
@@ -125,6 +134,36 @@ export class RestApi {
         return Promise.resolve(deferred);
     }
 
+    /**
+     * Determine the Compass version we're talking to.
+     * When completed, {@link _contentType} and {@link _user} are filled.
+     */
+    private async determineApiVersion(): Promise<any> {
```

Comment:
Oooone more thing; I think this is the first time we start using async / await. This is a relatively new JS function.

Don't know about any of our dependencies, but if not strictly necessary; perhaps better to stick with the old promises just a bit longer?
Afaik; client can just use 'await' on a function that returns a promise.

Reply:
This is being transpiled into non-async functions.
Check the generated `build/Compass.js`, all async stuff has been replaced by magic.